File: swex/add-in/export-components/vb-net/ComponentExtension.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System.Runtime.CompilerServices

Module ComponentExtension

    <Extension()>
    Sub Export(ByVal comp As IComponent2, ByVal filePath As String)

        Dim model = comp.IGetModelDoc()

        If model IsNot Nothing Then
            Dim err As Integer = -1
            Dim warn As Integer = -1
            model.Extension.SaveAs(filePath, CInt(swSaveAsVersion_e.swSaveAsCurrentVersion),
                                       CInt(swSaveAsOptions_e.swSaveAsOptions_Silent), Nothing, err, warn)
        Else
            Throw New NullReferenceException("Model")
        End If

    End Sub

End Module

File: swex/add-in/export-components/csharp/ComponentExtension.cs
using SolidWorks.Interop.swconst;
using System;

namespace SolidWorks.Interop.sldworks
{
    public static class ComponentExtension
    {
        public static void Export(this IComponent2 comp, string filePath)
        {
            var model = comp.IGetModelDoc();

            if (model != null)
            {
                int err = -1;
                int warn = -1;
                
                model.Extension.SaveAs(filePath, (int)swSaveAsVersion_e.swSaveAsCurrentVersion,
                    (int)swSaveAsOptions_e.swSaveAsOptions_Silent, null, ref err, ref warn);
            }
            else
            {
                throw new NullReferenceException("Model");
            }
        }
    }
}
